modules/sharedaddy/sharing.js

```diff
@@ -1,364 +1,474 @@
+			val
+		);
+	}
+
+	function closeEmailDialog() {
+		var dialog = document.querySelector( '#sharing_email' );
+		hideNode( dialog );
+	}
+
+	// Sharing initialization.
+	// Will run immediately or on `DOMContentLoaded`, depending on current page status.
+	function init() {
+		// Move email dialog to end of body.
+		var emailDialog = document.querySelector( '#sharing_email' );
+		document.body.appendChild( emailDialog );
+
+		WPCOMSharing_do();
+	}
+	if ( document.readyState !== 'loading' ) {
+		init();
+	} else {
+		document.addEventListener( 'DOMContentLoaded', init );
+	}
+
+	// Set up sharing again whenever a new post loads, to pick up any new buttons.
+	document.body.addEventListener( 'post-load', WPCOMSharing_do );
+
+	// Set up sharing, updating counts and adding all button functionality.
+	function WPCOMSharing_do() {
+		if ( window.WPCOMSharing ) {
+			window.WPCOMSharing.get_counts();
 		}
 
-		$( document ).click( function () {
-			// Click outside
-			// remove any timer
-			$more_sharing_buttons.each( function () {
-				clearTimeout( $( this ).data( 'timer' ) );
-			} );
-			$more_sharing_buttons.data( 'timer', false );
+		forEachNode( document.querySelectorAll( '.sharedaddy a' ), function ( anchor ) {
+			var href = anchor.getAttribute( 'href' );
+			if ( href && href.indexOf( 'share=' ) !== -1 && href.indexOf( '&nb=1' ) === -1 ) {
+				anchor.setAttribute( 'href', href + '&nb=1' );
+			}
+		} );
+
+		// Show hidden buttons
+
+		// Touchscreen device: use click.
+		// Non-touchscreen device: use click if not already appearing due to a hover event
 
-			// slide down forcibly
-			$( '.sharedaddy .inner' ).slideUp();
+		forEachNode( document.querySelectorAll( '.sharedaddy a.sharing-anchor' ), function (
+			buttonEl
+		) {
+			MoreButton.instantiateOrReuse( buttonEl );
 		} );
 
+		if ( document.ontouchstart !== undefined ) {
+			document.body.classList.add( 'jp-sharing-input-touch' );
+		}
+
 		// Add click functionality
-		$( '.sharedaddy ul' ).each( function () {
-			if ( 'yep' === $( this ).data( 'has-click-events' ) ) {
+		forEachNode( document.querySelectorAll( '.sharedaddy ul' ), function ( group ) {
+			if ( group.getAttribute( 'data-sharing-events-added' ) === 'true' ) {
 				return;
 			}
-			$( this ).data( 'has-click-events', 'yep' );
+			group.setAttribute( 'data-sharing-events-added', 'true' );
 
 			var printUrl = function ( uniqueId, urlToPrint ) {
-				$( 'body:first' ).append(
-					'<iframe style="position:fixed;top:100;left:100;height:1px;width:1px;border:none;" id="printFrame-' +
-						uniqueId +
-						'" name="printFrame-' +
-						uniqueId +
-						'" src="' +
-						urlToPrint +
-						'" onload="frames[\'printFrame-' +
-						uniqueId +
-						"'].focus();frames['printFrame-" +
-						uniqueId +
-						'\'].print();"></iframe>'
-				);
+				var iframe = document.createElement( 'iframe' );
+				iframe.style = 'position:fixed; top:100; left:100; height:1px; width:1px; border:none;';
```

Comment:
Does this really work? I thought you had to set style to an object.

Reply:
I think for a lot of these, you really want to be calling `iframe.setAttribute` rather than `iframe.attrname = 'foo'`. Don't know what the side effect of setting these as properties might be.

Reply:
I'll double-check all of this in a few browsers 👍

Reply:
Note: this seems to work in every browser I've tried. What we're doing here is setting the "style" property on the iframe element. Since DOM properties are strings, this works exactly as intended.

Reply:
Reading over the spec, it's probably best to set this as an attribute, though. The property is supposed to be an object, and the attribute is supposed to be a string. The fact that setting the property to a string works is probably reliable and probably spec-compliant, but it's not the best way of doing things.